ldap/servers/slapd/opshared.c

```diff
@@ -696,15 +696,15 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
              * In async paged result case, the search result might be released
              * by other theads.  We need to double check it in the locked region.
              */
-            pthread_mutex_lock(&(pb_conn->c_mutex));
+            pthread_mutex_lock(pageresult_lock_get_addr(pb_conn));
```

Comment:
micro-optimisation - keep a reference to the lock pointer from this point to re-use lower to avoid the extra dereferences.

Reply:
Done.